hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl

```diff
@@ -784,7 +798,94 @@
         { bits: "2",    name: "prog_full",  desc: "Flash program FIFO full"},
         { bits: "3",    name: "prog_empty", desc: "Flash program FIFO empty, software must provide data", resval: "1"},
         { bits: "4",    name: "init_wip",   desc: "Flash controller undergoing init, inclusive of phy init"},
-        { bits: "${8 + bank_width + page_width -1}:8", name: "error_addr", desc: "Flash controller error address."},
+      ]
+    },
+
+    { name: "ERR_CODE",
+      desc: '''
+        Flash error code register.
+        This register tabulates detailed error status of the flash.
+        This is separate from !!OP_STATUS, which is used to indicate the current state of the software initiated
+        flash operation.
+      '''
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "flash_err",
+          desc: '''
+            The flash memory itself has an error, please check the vendor specs for details of the error.
+          '''
+        },
+        { bits: "1",
+          name: "flash_alert",
+          desc: '''
+            The flash memory itself has triggered an alert, please check the vendor specs for details of the error.
+          '''
+        },
+        { bits: "2",
+          name: "mp_err",
+          desc: '''
+            Flash access has encountered an access permission error.
+            Please see !!ERR_ADDR for exact address.
+          '''
+        },
+        { bits: "3",
+          name: "ecc_single_err",
+          desc: '''
+            Flash access has encountered a recoverable ECC error.
```

Comment:
should we reword the ECC errors to "correctable" and "uncorrectable" in order to not confuse them with the alerts?
i.e., at the moment we have "non-recoverable" ECC errors that can still trigger a "recoverable" alert, and that may be a bit confusing...

Reply:
i actually intentionally named it this way because from software perspective, they can technically recover from both.  One they don't need to do something (but probably should) and the other they definitely need to do something.

That's the main reason i just went for single and multi and didn't try to dictate in the description whether it's correctable or not.

Reply:
Yes, that's true. And I think the `ecc_single_err`/`ecc_multi_err` names are fine. 
I was more referring to the description of the `ecc_multi_err` which says it is a non-recoverable ECC error (yet, the alert is recoverable and as you mentioned, SW can technically recover from it).
How about we use the terms `single bit ECC error` and `multi bit ECC error` in the descriptions instead of `recoverable`/`non-recoverable`?